lib/recommendation-server.js

```diff
@@ -55,6 +57,39 @@ RecommendationServer.prototype = {
     const endpoint = useDummyEndpoint ? 'dummy' : '';
     return `${this.server}/${endpoint}?q=${encodeURIComponent(query)}`;
   },
+
+  _resultHasChanged(data) {
+    /*
```

Comment:
Took some serious digging in dxr, but it looks like you can load `ObjectUtils.jsm` and call `ObjectUtils.deepEqual()` to compare two objects. That should simplify this function quite a bit :-)